lib/mongo/collection/view/iterable.rb

```diff
@@ -34,24 +34,49 @@ module Iterable
         # @since 2.0.0
         #
         # @yieldparam [ Hash ] Each matching document.
+
         def each
-          @cursor = nil
-          session = client.send(:get_session, @options)
-          @cursor = if respond_to?(:write?, true) && write?
-            server = server_selector.select_server(cluster, nil, session)
-            result = send_initial_query(server, session)
-            Cursor.new(view, result, server, session: session)
-          else
-            read_with_retry_cursor(session, server_selector, view) do |server|
-              send_initial_query(server, session)
+          if system_collection? || !QueryCache.enabled?
```

Comment:
Can this condition be inverted such that the "if" branch is the one with the cache enabled and the "else" branch is the one without the cache enabled please.